### DISCUSSION TITLE
Generating model grids

### DISCUSSION THREAD

**Discussion by s-goldman** [dullemond/radmc3d-2.0#4] [General]
I am looking for a way to use RADMC3D to create a grid of simple realistic radiative transfer models for evolved stars. From what I understand, the package isn't really designed for this. Before I spend too much time trying to get it to work, is this something that is possible?

**Comment by dullemond** [comment]
Interesting question. If you are interested in, for instance, the dusty winds of AGB stars, then yes, RADMC-3D can do this. If your star has a lot of atomic or molecular lines, then one would need methods such as 'correlated k' opacities. This is not (yet) implemented into RADMC-3D. Strictly speaking this is technically not much different from any other LTE-type radiative transfer, and with some work one could let RADMC-3D do the ray-tracing part of that. To actually do the thermal equilibrium with 'correlated k' is a bit more tricky, and would require some adaption to RADMC-3D. The reason is that RADMC-3D does thermal equilbrium using dust opacities. These are opacities that are independent of density and temperature. This allows RADMC-3D to use the Björkman & Wood algorithm. For gas opacities (with 'correlated k' method) the opacities depend on density and temperature, and the Björkman & Wood algorithm can not be used. Instead, the Lucy algorithm should be used. That is not (yet) implemented, although it should not be hard to do (it's simpler than the Björkman & Wood algorithm). All of this is possible, but requires some work. So it depends: if you find another code that already has this stuff, I recommend you to use that one. If not, then I'd be happy to help you adapt RADMC-3D. Getting the appropriate 'correlated k' opacity tables is, however, not trivial. The petitRADTRANS code https://gitlab.com/mauricemolli/petitRADTRANS of Paul Molliere has computed large grids of low-temperature 'correlated k' opacity tables. But I am not sure if they reach to the density regime you need them for. 

So the short answer is: presumably not. But it depends on what exactly you want to do.

**Reply by s-goldman** [reply]
@dullemond I am in fact interested in the dusty winds of AGB stars, and I am more interested in the continuum. If you have any suggestions on where to start on a project like this (other than the documentation) that would be really helpful. Also, if there is anyone else reading this that is interested in (or has experience with) creating these types of models, please feel free to message me.

**Comment by dullemond** [comment]
This should be very much doable, as this is precisely the kind of models for which RADMC-3D was built: circumstellar dusty envelopes. But with RADMC-3D everything requires some hands-on work, as the code is not meant as a "model" but as a "modeling tool". I guess you may want to start with a 1-D spherically symmetric model (see examples/run_spher1d_1/ to start with). You'd presumably have to also set up your choice of dust composition (see opac/dust_continuum/). Once that all works, you could try to move to 2-D spherical coordinates (examples/run_spher2d_1/) and take it from there.

**Comment by s-goldman** [comment]
That's great to hear, thanks!
